### ISSUE TITLE
Bug in default database name

### ISSUE THREAD

**Issue by Michiel-s** [AmpersandTarski/Ampersand#598]
When generating a prototype the filename (without extension) of the entry script is used as database name. This can lead to a database error during installation when using names that are not allowed as database name.

This issue request to change this default behavior by using the name of the CONTEXT of the entry script instead, prefixed with 'ampersand_'

**Comment by hanjoosten** [comment]
Whontfix:

There are multiple ways you can influence the name of the database: You can use the --dbName switch. Using THE context name isn't a good solution too, because we will be able to define scripts with multiple contexts in the (near?) future. Therefor, the name of the script makes a less bad candidte as databasename than the contextname. If you don't like it, you can use the switch. And you can use the include.yaml stuff to make that semi-permanent for your app.

**Comment by RieksJ** [comment]
While the proposed solution is obviously debatable, this issue itself is not. 

In order to avoid that some of us will have to deal with issues of (predominantly novice) users that would not guess that filenames are used as database names, it would be nice to at least make sure that the generated database name is actually a valid one.

**Comment by hanjoosten** [comment]
good point. So whatever the dbName would be, we need to check that it matches the [restrictions MySQL has for database names](http://dev.mysql.com/doc/refman/5.7/en/identifiers.html).

**Comment by hanjoosten** [comment]
This also goes for the names of tables and columns. Currently the script in [ampersand/testing/Sentinel/Tests/ShouldSucceed/Ticket580.adl](https://github.com/AmpersandTarski/Ampersand/blob/development/testing/Sentinel/Tests/ShouldSucceed/Ticket580.adl) fails because of a naming issue (fancy character in table name):

```
PHP/SQL problem: Error : Connect to server failed1146: Table 'TempDB_ampersand_ticket580.NormenhiÃ«rarchie' doesn't exist(Sql: $sql)
```

**Comment by stefjoosten** [comment]
What precisely are the constraints to names of database, database tables and columns in SQL? Please refer to the authoritative source (some SQL standard).

**Comment by hanjoosten** [comment]
@stefjoosten see [this comment](https://github.com/AmpersandTarski/Ampersand/issues/598#issuecomment-268450008) above, which contains the reference.

**Comment by Michiel-s** [comment]
We’ll fix this in #708
